tests/4.5/target_teams_distribute/test_target_teams_distribute_default_shared.F90

```diff
@@ -0,0 +1,84 @@
+!===--- test_target_teams_distribute_default_shared.F90---------------------===//
+!
+! OpenMP API Version 4.5 Nov 2015
+!
+! This test uses the default(shared) clause on a target teams distribute
+! directive.  The test aims to validate that when the default(shared) clause
+! is present, that all variables without explicit data sharing attributes
+! will be shared within the region.  To test this, we test that a data element
+! that should be shared due to the default(shared) clause is available to
+! all the teams.  The first test uses atomic to write to the variable without
+! race conditions.  The second test uses synchronization constructs to have
+! one thread change the shared variable and ensures all threads see the change.
+!
+!===------------------------------------------------------------------------===//
+
+#include "ompvv.F90"
+
+#define N 1024
+
+PROGRAM test_target_teams_distribute_default_shared
+  USE iso_fortran_env
+  USE ompvv_lib
+  USE omp_lib
+  implicit none
+  INTEGER :: errors
+  OMPVV_TEST_OFFLOADING
+  errors = 0
+
+  OMPVV_TEST_VERBOSE(default_shared1() .ne. 0)
+  OMPVV_TEST_VERBOSE(default_shared2() .ne. 0)
+
+  OMPVV_REPORT_AND_RETURN()
+
+CONTAINS
+  INTEGER FUNCTION default_shared1()
+    INTEGER :: a(N)
+    INTEGER :: share, errors, num_teams, x
+    errors = 0
+    share = 0
+
+    DO x = 1, N
+       a(x) = x
+    END DO
+
+    !$omp target data map(to: a(1:N)) map(tofrom: share, num_teams)
+    !$omp target teams distribute default(shared) defaultmap(tofrom:scalar)
+    DO x = 1, N
+       num_teams = omp_get_num_teams()
```

Comment:
Data race, guard num_teams

Reply:
Num teams is not used afterwards

Reply:
Fixed with team num check, added warning below which uses the value.